themes/default/content/blog/api-gateway-to-eventbridge/index.md

```diff
@@ -0,0 +1,371 @@
+
+With the event bus and event rule in place, you’re ready to define the integration itself — the resource responsible for connecting the gateway route (which we’ll get back to shortly) to your newly created event bus. As I mentioned earlier, there are several types of API Gateway integration to choose from, each one suited to a particular purpose. For this example, the `AWS_PROXY` type is good fit, as it’s simple and requires very little code; it doesn't give you quite as much control over the API Gateway response as you might like --- as a proxy, it just returns to the caller whatever is returned by the backend, in this case EventBridge --- but it's more than enough for the task at hand.
+
+Add the following lines for the integration and route. The comments should explain what each block is doing:
+
+```typescript
+...
+
+// Define a policy granting API Gateway permission to publish to EventBridge.
+const apiGatewayRole = new aws.iam.Role("api-gateway-role",
+    {
+        assumeRolePolicy: {
+            Version: "2012-10-17",
+            Statement: [
+                {
+                    Action: "sts:AssumeRole",
+                    Effect: "Allow",
+                    Principal: {
+                        Service: "apigateway.amazonaws.com",
+                    },
+                },
+            ],
+        },
+        managedPolicyArns: [
+            "arn:aws:iam::aws:policy/AmazonEventBridgeFullAccess",
+        ],
+    },
+);
+
+// Create an API Gateway integration to forward requests to EventBridge.
+const integration = new aws.apigatewayv2.Integration("integration", {
+    apiId: api.id,
+
+    // The integration type and subtype.
+    integrationType: "AWS_PROXY",
+    integrationSubtype: "EventBridge-PutEvents",
+    credentialsArn: apiGatewayRole.arn,
+
+    // The body of the request to be sent to EventBridge. Note the
+    // event source matches pattern defined on the EventRule, and the
+    // Detail expression, which just forwards the body of the original
+    // API Gateway request (i.e., the uploaded document).
+    requestParameters: {
+        EventBusName: bus.name,
+        Source: "my-event-source",
+        DetailType: "my-detail-type",
+        Detail: "$request.body",
+    },
+});
+
+// Finally, define the route.
+const route = new aws.apigatewayv2.Route("route", {
+    apiId: api.id,
+    routeKey: "POST /uploads”,
+    target: pulumi.interpolate`integrations/${integration.id}`,
+});
+```
+
+With that, you’re ready to configure the Lambda.
+
+## Add a Lambda function handler
+
+The hard part’s done: you’ve declared an API and route, mapped that route to an integration, configured the integration to publish events to an event bus, and defined an event rule to respond to those events. All that's left now is to tell the rule _how_ to respond.
```

Comment:
```suggestion
The hard part’s done: You’ve declared an API and route, mapped that route to an integration, configured the integration to publish events to an event bus, and defined an event rule to respond to those events. All that's left now is to tell the rule _how_ to respond.
```

Reply:
Keeping this one also since I believe this is technically okay to use either one.